internal/system/agent/getexecutor.go

```diff
@@ -0,0 +1,107 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/startup"
+	"github.com/edgexfoundry/edgex-go/internal/system/agent/response"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/general"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
+	"github.com/edgexfoundry/go-mod-registry/registry"
+)
+
+type getExecutor struct {
+	genClients      *GeneralClients
+	configClients   ConfigurationClients
+	registryClient  registry.Client
+	loggingClient   logger.LoggingClient
+	serviceProtocol string
+}
+
+func NewGetExecutor(
+	genClients *GeneralClients,
+	configClients ConfigurationClients,
+	registryClient registry.Client,
+	loggingClient logger.LoggingClient,
+	serviceProtocol string) *getExecutor {
+
+	return &getExecutor{
+		genClients:      genClients,
+		configClients:   configClients,
+		registryClient:  registryClient,
+		loggingClient:   loggingClient,
+		serviceProtocol: serviceProtocol,
+	}
+}
+
+// Get provides the common callout to the config service executor.
+func (ce getExecutor) Get(services []string, ctx context.Context) interface{} {
+	result := struct {
+		Configuration map[string]interface{} `json:"configuration"`
+	}{
+		Configuration: map[string]interface{}{},
+	}
+
+	// Loop over services and accumulate the response (i.e. "result") to return to requester.
+	for _, service := range services {
+		config, err := ce.getConfig(service, ctx)
+		if err != nil {
+			ce.loggingClient.Error(fmt.Sprintf(err.Error()))
+			result.Configuration[service] = fmt.Sprintf(err.Error())
+			continue
+		}
+		ce.loggingClient.Debug(fmt.Sprintf("result in Do: %v", config))
+		result.Configuration[service] = response.Process(config, ce.loggingClient)
+	}
+	return result
+}
+
+func (ce getExecutor) getConfig(serviceName string, ctx context.Context) (string, error) {
+	var result string
+	client, ok := ce.genClients.Get(serviceName)
+	if !ok {
+		if ce.registryClient == nil {
+			return "", fmt.Errorf("registryClient not initialized; required to handle unknown service: %s", serviceName)
+		}
+
+		// Service unknown to SMA, so ask the Registry whether `serviceName` is available.
+		if err := ce.registryClient.IsServiceAvailable(serviceName); err != nil {
+			return "", err
+		}
+
+		ce.loggingClient.Info(fmt.Sprintf("Registry responded with %s available", serviceName))
+
+		// Since serviceName is unknown to SMA, ask the Registry for a ServiceEndpoint associated with `serviceName`
+		endpoint, err := ce.registryClient.GetServiceEndpoint(serviceName)
+		if err != nil {
+			return "", fmt.Errorf("on attempting to get ServiceEndpoint for %s, got error: %v", serviceName, err.Error())
+		}
+
```

Comment:
I removed the following logical check because we handle the corresponding handling further upstream in this function:
```
        usingRegistry := false
        if ce.registryClient != nil {
          usingRegistry = true
        }
```
i.e. we handle it here:
```
		if ce.registryClient == nil {
			return "", fmt.Errorf("registryClient not initialized; required to handle unknown service: %s", serviceName)
		}
```